db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/OptimizerImpl.java

```diff
@@ -2152,13 +2173,23 @@ private void pullOptimizableFromJoinOrder() throws StandardException{
         assignedTableMap.xor(pullMe.getReferencedTableMap());
     }
 
+    /* This check may have other side effects but we couldn't find
+     * a better way at this point. Problems:
+     * 1) The same statement, when executed at a different time (being
+     *    the first statement or executed after some other statements)
+     *    , may have 0 or 1 returned from lcc.getActivationCount().
+     * 2) Not all kinds of statement have a ResultDescription.
+     */
     private boolean isPreparedStatement() {
         LanguageConnectionContext lcc =
             (LanguageConnectionContext) ContextService.
                 getContextOrNull(LanguageConnectionContext.CONTEXT_ID);
         if (lcc == null || lcc.getActivationCount() < 1)
             return false;
-        return lcc.getLastActivation().getPreparedStatement() != null;
+        Activation ac = lcc.getLastActivation();
+        ResultDescription rd = ac.getResultDescription();
```

Comment:
I don't see the assertion in BaseActivation.getResultDescription removed.  Was this change not required after all?

Reply:
It is removed. Please check `BaseActivation.java` in this change.

Reply:
OK, thanks.